src/sources/http_scrape/tests.rs

```diff
@@ -0,0 +1,244 @@
+use std::collections::HashMap;
+use tokio::time::Duration;
+use warp::{http::HeaderMap, Filter};
+
+use crate::{serde::default_decoding, serde::default_framing_message_based};
+use codecs::decoding::{
+    CharacterDelimitedDecoderOptions, DeserializerConfig, FramingConfig,
+    NewlineDelimitedDecoderOptions,
+};
+use vector_core::event::Event;
+
+use super::HttpScrapeConfig;
+use crate::test_util::{
+    components::{run_and_assert_source_compliance, HTTP_PULL_SOURCE_TAGS},
+    next_addr, test_generate_config, wait_for_tcp,
+};
+
+pub(crate) const INTERVAL_SECS: u64 = 1;
+
+/// The happy path should yield at least one event and must emit the required internal events for sources.
+pub(crate) async fn run_compliance(config: HttpScrapeConfig) -> Vec<Event> {
+    let events =
+        run_and_assert_source_compliance(config, Duration::from_secs(3), &HTTP_PULL_SOURCE_TAGS)
+            .await;
+
+    assert!(!events.is_empty());
+
+    events
+}
+
+#[test]
+fn http_scrape_generate_config() {
+    test_generate_config::<HttpScrapeConfig>();
+}
+
+/// Bytes should be decoded and HTTP header set to text/plain.
```

Comment:
Tiny nit (non blocking): Test functions don't need doc comments (`///`), as they will never be rendered.

Reply:
Ah good to know... is it ok to keep as-is though? I like to use them as alternatives to inline comments as the editor renders them in different colors. Makes it a little easier for the eyes to spot structs, functions etc. :shrug: 
But I'm happy to remove them if its an inconsistency with the project etc- I guess that is the bigger question. If no one uses doc comments on test functions, it makes sense to change them to inline.

Reply:
No, no need to change them, was just noting that they are unnecessary.